pilot/pkg/model/telemetry.go

```diff
@@ -454,6 +469,13 @@ func mergeLogs(logs []*tpb.AccessLogging, mesh *meshconfig.MeshConfig) (sets.Set
 				providers.Delete(provider)
 				continue
 			}
+
+			if m.Match != nil &&
```

Comment:
do we need to move this before `providers.Delete(provider)`

Reply:
It feels like it does. @zirain ? I think this means we also need a test for:

- mesh-wide: enabled
- ns-wide: disabled

and 

- mesh-wide: enabled
- ns-wide: disabled
- workload: enabled

just to validate we are deleting and re-adding appropriately.